radixdlt-core/radixdlt/src/main/java/com/radixdlt/statecomputer/RadixEngineStateComputer.java

```diff
@@ -395,27 +395,7 @@ public StateComputerResult prepare(
       throw new ByzantineQuorumException(e.getMessage(), e);
     }
 
-    result
-        .getMetadata()
-        .getNextForkHash()
-        .ifPresent(
-            nextForkHash -> {
-              final var nextForkConfig =
-                  forks.getByHash(nextForkHash).orElseThrow(); // guaranteed to be present
-              log.info(
-                  "Epoch {} forking RadixEngine to {}",
-                  proof.getEpoch() + 1,
-                  nextForkConfig.name());
-              final var rules = nextForkConfig.engineRules();
-              this.radixEngine.replaceConstraintMachine(
-                  rules.getConstraintMachineConfig(),
-                  rules.getSerialization(),
-                  rules.getActionConstructors(),
-                  rules.getPostProcessor(),
-                  rules.getParser());
-              this.epochCeilingView = rules.getMaxRounds();
-              this.maxSigsPerRound = rules.getMaxSigsPerRound();
-            });
+    result.getMetadata().getNextForkName().ifPresent(this::forkRadixEngine);
```

Comment:
I don't really have a suggestion, but it seems a bit weird this is mutating state. What do you think?

Reply:
what do you mean? the state is usually supposed to be modified ;)